packages/SwingSet/src/kernel/state/vatKeeper.js

```diff
@@ -179,6 +179,18 @@ export function makeVatKeeper(
     }
   }
 
+  function markAsDead() {
+    const deathKey = `${vatID}.dead`;
+    if (storage.has(deathKey)) {
```

Comment:
This sounds fine as a safety check, but it makes me wonder what would actually happen if the userspace holder of the vat's control facet were to call `terminate` twice. I think it would cause a kernel panic, because it'd be a device making the call that eventually hits this Error, which qualifies as an internal syscall error, which is kernel-fatal.

I think we need to add a flag in the control facet to throw an already-terminated error earlier, before the kernel function is invoked. And of course a test for this case.

Reply:
I just now looked at how control gets to `markAsDead.  My assessment is that there should be a check in `terminateVat` to test if the vat is dead before initiating any cleanup work at all.  It's not obvious to me if the correct behavior in response to finding out that the vat is already dead should be to throw an error back to the user or simply to silently accept the call and do nothing -- hey, you wanted this vat dead, well it is so we're good.  I'm inclined towards the latter -- it's the same reasoning as why removing a non-existent entry from a map is benign and why you can delete a non-existent file.

Reply:
yeah, I'm ok with the userspace call to `controlFacet.terminate()` just being ignored, as long as it doesn't cause a kernel panic